functional_tests/steps/bundles.py

```diff
@@ -121,3 +128,272 @@ def the_user_can_see_the_bundle_details_with_creator(context: Context) -> None:
+
+
+def add_reslease_calendar_in_edit(context: Context) -> None:
+    # add Release Calendar
+    if hasattr(context, "search_release_calendar"):
+        expect(context.page.get_by_role("button", name="Choose Release Calendar page")).to_be_visible()
+        context.page.get_by_role("button", name="Choose Release Calendar page").click()
+        context.page.get_by_role("textbox", name="Search term").click()
+        context.page.get_by_role("textbox", name="Search term").fill(context.release_calendar_pages[0].title)
+        expect(context.page.get_by_role("link", name=context.release_calendar_pages[0].title)).to_be_visible()
+        context.page.get_by_role("row", name=context.release_calendar_pages[0].title).get_by_role("link").click()
+        context.page.get_by_role("link", name=context.release_calendar_pages[0].title).click()
+
+
+@step("the user can preview a bundle")
+def can_preview_bundle(context: Context) -> None:
+    expect(context.page.locator("#bundles-for-preview-heading")).to_contain_text("Bundles ready for preview")
+    expect(context.page.locator("#bundles-for-preview-content")).to_contain_text(context.bundles[0].name)
+    context.page.get_by_title("View").click()
+
+    expect(context.page.get_by_text("Name")).to_be_visible()
+    expect(context.page.get_by_text("Created at")).to_be_visible()
+    expect(context.page.get_by_text("Created by")).to_be_visible()
+    expect(context.page.get_by_text("Approval status")).to_be_visible()
+    expect(context.page.get_by_text("Associated release calendar")).to_be_visible()
+    expect(context.page.get_by_text("Teams", exact=True)).to_be_visible()
+    expect(context.page.locator("#main").get_by_text("Pages")).to_be_visible()
+    expect(context.page.get_by_text("Datasets", exact=True)).to_be_visible()
+
+
+@step("the user cannot preview a bundle")
+def cannot_preview_bundle(context: Context) -> None:
+    context.page.get_by_role("link", name="Bundles", exact=True).click()
+    expect(context.page.get_by_role("button", name=f"More options for '{context.bundles[0].name}'")).not_to_be_visible()
+
+
+@step("the user cannot approve a bundle")
+def cannot_approve_bundle(context: Context) -> None:
+    context.page.get_by_role("link", name="Bundles", exact=True).click()
+    expect(context.page.get_by_role("button", name=f"More options for '{context.bundles[0].name}'")).not_to_be_visible()
+
+
+@step("the user can approve a bundle")
+def can_approve_bundle(context: Context) -> None:
+    if len(context.bundles) == 1:
+        expect(context.page.locator("#latest-bundles-heading")).to_contain_text("Latest active bundles")
+        expect(context.page.locator("#latest-bundles-content")).to_contain_text(context.bundles[0].name)
+        context.page.get_by_title("Edit this bundle").click()
+        context.page.get_by_role("button", name="More actions").click()
+        context.page.get_by_role("button", name="Approve").click()
```

Comment:
You can take these two lines out of the if/else block so that you don't repeat yourself